workflows/sagemaker_pipelines/SPECIFICATION.md

```diff
@@ -0,0 +1,331 @@
+https://docs.aws.amazon.com/sagemaker/latest/dg/pipelines-studio.html
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+Finally, let's clean up our resources.
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+Clean up resources.
+        Delete pipeline sagemaker-sdk-example-pipeline? (y/n)
+y
+        Delete queue https://sqs.us-west-2.amazonaws.com/565846806325/sagemaker-sdk-example-queue-rlhagerm? (y/n)
+y
+        Delete Amazon S3 bucket sagemaker-sdk-test-bucket-rlhagerm2? (y/n)
+y
+        Delete role SageMakerExampleLambdaRole? (y/n)
+y
+        Delete role SageMakerExampleRole? (y/n)
+y
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+SageMaker pipeline scenario is complete.
+--------------------------------------------------------------------------------
+```
+### Hello SageMaker
+The Hello Service example should demonstrate how to set up the client and make an example call using the SDK.
+
+Initialize the client and call ListNotebookInstances to list up to 5 of the account's notebook instances. If no instances are found, you can direct the user to instructions on how to add one.
+
+Sample output:
+
+```
+Hello Amazon SageMaker! Let's list some of your notebook instances:
+    Instance: test-notebook
+    Arn: arn:aws:sagemaker:us-west-2:123456789:notebook-instance/test-notebook
+    Creation Date: 6/7/2023
+```
+
+_General info for Hello Service example snippets:_
+This section of the workflow should be a streamlined, simple example with enough detail to be as close to “copy/paste” runnable as possible. This example may include namespaces and other setup in order to focus on getting the user up and running with the new service.
+
+### README
+
+This is a workflow scenario. As such, the READMEs should be standardized.
+This is the [.NET reference README](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/dotnetv3/SageMaker/Scenarios/README.md). When a language implementation is completed, update the [parent README](README.md) to include the new language SDK.
+
+## Troubleshooting
+- You may want to view your pipeline in SageMaker studio, which will require a Domain
+This will provide better debugging for the pipeline execution steps. You can use a default Domain or create a custom Domain.
+- Cloudwatch logs will be your friend. SageMaker studio should link each step to relevant logs.
+- When testing your Lambda function, you may find it useful to log out the function input. Serialization differences between languages (capitalization, nulls) can cause the function to fail.
+- When pipelines are failing, don't delete the queue until the pipeline execution has stopped.
```

Comment:
Can we add a note about callbacks never resolving until `SendPipelineExecutionStepSuccess` or `SendPipelineExecutionStepFailure` are called. There's a risk of having to reach out to support if these aren't called.

Reply:
Added.